context/kuberetes_client.go

```diff
@@ -50,22 +50,41 @@ func (c *KubernetesClient) SetExpiry(def time.Duration) {
 
 func (c *KubernetesClient) Refresh(ctx Context) error {
 	if !c.HasExpired() {
-		c.logger.Tracef("Skipping refresh, client has not expired")
+		c.logger.Infof("Skipping refresh, client has not expired %s", c.Config.Host)
 		return nil
 	}
-	_, rc, err := c.Connection.Populate(ctx, true)
+	client, rc, err := c.Connection.Populate(ctx, true)
 	if err != nil {
 		return fmt.Errorf("error refreshing kubernetes client: %w", err)
 	}
 
+	doRefresh := false
 	// Update rest config in place for easy reuse
 	c.Config.Host = rc.Host
 	c.Config.TLSClientConfig = rc.TLSClientConfig
-	c.Config.BearerToken = rc.BearerToken
+	if c.Config.BearerToken != rc.BearerToken {
+		logger.Infof("=== BEARER TOKEN UPDATED FOR host: %s", c.Config.Host)
+		doRefresh = true
+		c.Config.BearerToken = rc.BearerToken
+	}
+	if c.Config.Password != rc.Password {
+		doRefresh = true
+	}
+
 	c.Config.BearerTokenFile = rc.BearerTokenFile
 	c.Config.Username = rc.Username
 	c.Config.Password = rc.Password
 
+	logger.Infof("Host %s", rc.Host)
+	logger.Infof("BearerToken %+v", rc.BearerToken)
+	logger.Infof("Address of bt in Refresh is %p", &c.Client.Config.BearerToken)
+
+	if doRefresh {
+		logger.Infof("Refreshing client for host %s", rc.Host)
+		c.Client.Interface = client
```

Comment:
I think we can do this always as the client as already been created and is pretty cheap.

Reply:
Also https://github.com/flanksource/duty/blob/main/kubernetes/k8s.go#L55 that kube-client-cache is not expiry aware, so we could ask for a new client and it returns something that is 30m old that has already expired

Reply:
That cache has 1 hour expiry, should I update it to never use the cache itself ?

Reply:
Remove the kubeconfig based cache